src/app/cake/cake.service.ts

```diff
@@ -0,0 +1,24 @@
+import { Connection } from 'mongoose';
+import { iCake } from './cake.interface';
+import { CakeSchema } from './cake.model';
+import { DataService } from './../components/data-service.component';
+
+export class CakeService {
+  private dataService: DataService<iCake>;
+
+  constructor(dbConn: Connection) {
+    this.dataService = new DataService(dbConn, 'Cake', CakeSchema);
```

Comment:
Maybe moving the hard codded `Cake` to the model file as `export const modelName: string  = 'Cake'`, would be a better idea 😅

Reply:
Sure, thanks!